### PR TITLE
Documentation: update go tutorials

### PR DESCRIPTION
**Pull request by stevenferrer**
- Updated Go tutorials for the latest version of Tendermint, which is `v.0.35.8`

cc: @sergio-mena

### CODE REVIEW

File: docs/tutorials/go.md [hunk 1]
```diff
@@ -150,10 +152,34 @@ func (KVStoreApplication) ApplySnapshotChunk(abcitypes.RequestApplySnapshotChunk
...
+}
+
+func NewKVStoreApplication(db *badger.DB) *KVStoreApplication {
+ return &KVStoreApplication{
+  db: db,
+ }
+}
+```
+
+### 1.3.2 CheckTx
```
Comment (cmwaters) [inline docs/tutorials/go.md:182]: Would be nice to mention the notion of `priority` and `sender` here if possible

**Review by cmwaters — APPROVED** [review]
Thanks for helping out! Much appreciated.

File: docs/tutorials/go.md (hunk 1 above)
Comment (stevenferrer) [reply docs/tutorials/go.md:182]: Hi @cmwaters, I am still in the process of exploring the Tendermint ecosystem so I am not sure if I can document that properly. I was only aiming to update the Go tutorials so others can also benefit from what I learned so far.

Still, if you require it, I would try to explore about this topic and add it to the PR.

Comment (cmwaters) [reply docs/tutorials/go.md:182]: That's okay - we can just file an issue and follow up at a later point

Comment (stevenferrer) [reply docs/tutorials/go.md:182]: Got it, thanks @cmwaters!

**Pull request merged**